### PR TITLE
Explicit http verbs are needed when OpenAPI support is enabled

### PR DESCRIPTION
**Pull request by fiyazbinhasan**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/23248

@Rick-Anderson, I think a simple warning messege will suffice. 

[Internal Preview URL](https://review.docs.microsoft.com/en-us/aspnet/core/web-api/handle-errors?view=aspnetcore-6.0&branch=pr-en-us-24305).

### CODE REVIEW

File: aspnetcore/web-api/handle-errors.md [hunk 1]
```diff
@@ -91,6 +91,9 @@ The preceding `HandleError` action sends an [RFC 7807](https://tools.ietf.org/ht
 > [!WARNING]
 > Don't mark the error handler action method with HTTP method attributes, such as `HttpGet`. Explicit verbs prevent some requests from reaching the action method. Allow anonymous access to the method if unauthenticated users should see the error.
 
+> [!WARNING]
+> There is a caveat of having an action method without an HTTP method attribute specially if you have OpenAPI (Swagger) support enabled in your web API project. It will not be able to load the API definition in Swagger UI. In that case, it is suggested to ignore the action method from the API defination using an ignore attibute like, `[ApiExplorerSettings(IgnoreApi = true)]`
```
Comment (serpent5) [inline aspnetcore/web-api/handle-errors.md:95]: Let's combine the two warnings, as they're strongly related. I don't think we need to go into detail about Swashbuckle falling over if this attribute isn't used. Just recommending that it's used should cover it:

```suggestion
> [!WARNING]
> Don't mark the error handler action method with HTTP method attributes, such as `HttpGet`. Explicit verbs prevent some requests from reaching the action method.
>
> For web APIs that use [Swagger / OpenAPI](xref:tutorials/web-api-help-pages-using-swagger), mark the error handler action with the [[ApiExplorerSettings]](<xref:Microsoft.AspNetCore.Mvc.ApiExplorerSettingsAttribute>) attribute and set its <xref:Microsoft.AspNetCore.Mvc.ApiExplorerSettingsAttribute.IgnoreApi%2A> property to true. This attribute configuration excludes the error handler action from the OpenAPI specification for the app:
>
> ```csharp
> [ApiExplorerSettings(IgnoreApi = true)]
> ```
>
> Allow anonymous access to the method if unauthenticated users should see the error.
```

Comment (fiyazbinhasan) [reply aspnetcore/web-api/handle-errors.md]: looks great @serpent5! 👍

**Comment by serpent5** [comment]
@fiyazbinhasan As you've been helping out around here so much (thank you!), here's a few pointers I've picked up from @Rick-Anderson and the rest of the team:

- Avoid future tense. For example, instead of *It will not be able to load the API definition*, use something like *It's unable to load the API definition*.
- Use contractions. For example, instead of *There is a caveat*, use *There's a caveat*.
- Avoid "you" where possible. For example, instead of *if you have OpenAPI (Swagger) support enabled in your web API project*, use something like *For web API projects that use OpenAPI (Swagger), ...*.

I hope that's helpful. Just a few small changes like this make a big difference. 👍 Let me know if there's anything I can help you with as you continue to contribute.

File: aspnetcore/web-api/handle-errors.md (hunk 1 above)
Comment (serpent5) [reply aspnetcore/web-api/handle-errors.md]: Thanks @fiyazbinhasan. That's teamwork at its finest:

High Five

File: aspnetcore/web-api/handle-errors.md [hunk 2]
```diff
@@ -89,7 +89,15 @@ In non-development environments, use [Exception Handling Middleware](xref:fundam
...
-> Don't mark the error handler action method with HTTP method attributes, such as `HttpGet`. Explicit verbs prevent some requests from reaching the action method. Allow anonymous access to the method if unauthenticated users should see the error.
+> Don't mark the error handler action method with HTTP method attributes, such as `HttpGet`. Explicit verbs prevent some requests from reaching the action method.
+>
+> For web APIs that use [Swagger / OpenAPI](xref:tutorials/web-api-help-pages-using-swagger), mark the error handler action with the [[ApiExplorerSettings]](<xref:Microsoft.AspNetCore.Mvc.ApiExplorerSettingsAttribute>) attribute and set its <xref:Microsoft.AspNetCore.Mvc.ApiExplorerSettingsAttribute.IgnoreApi%2A> property to true. This attribute configuration excludes the error handler action from the OpenAPI specification for the app:
+>
+> ```csharp
+> [ApiExplorerSettings(IgnoreApi = true)]
+> ```
+>
+> Allow anonymous access to the method if unauthenticated users should see the error.
```
Comment (serpent5) [inline aspnetcore/web-api/handle-errors.md:100]:
```suggestion
> Don't mark the error handler action method with HTTP method attributes, such as `HttpGet`. Explicit verbs prevent some requests from reaching the action method.
>
> For web APIs that use [Swagger / OpenAPI](xref:tutorials/web-api-help-pages-using-swagger), mark the error handler action with the [[ApiExplorerSettings]](<xref:Microsoft.AspNetCore.Mvc.ApiExplorerSettingsAttribute>) attribute and set its <xref:Microsoft.AspNetCore.Mvc.ApiExplorerSettingsAttribute.IgnoreApi%2A> property to `true`. This attribute configuration excludes the error handler action from the app's OpenAPI specification:
>
> ```csharp
> [ApiExplorerSettings(IgnoreApi = true)]
> ```
>
> Allow anonymous access to the method if unauthenticated users should see the error.
```

**Comment by fiyazbinhasan** [comment]
Thanks, @serpent5. Those are some great pointers! I'll keep those in mind 🙂

**Pull request merged**